cmd/snap/cmd_advise.go

```diff
@@ -205,11 +210,50 @@ func adviseViaAptHook() error {
 	return nil
 }
 
+type Snap struct {
+	Snap    string
+	Version string
+	Command string
+}
+
+func dumpDbHook() error {
+	commands, err := advisor.DumpCommands()
+	if err != nil {
+		return err
+	}
+
+	commands_processed := make([]string, 0)
+	var b []Snap
+
+	for key, value := range commands {
+		err := json.Unmarshal([]byte(value), &b)
+		if err != nil {
+			return err
+		}
+		for i := range b {
+			var s = fmt.Sprintf("%s\xff%s/s\n", key, b[i].Snap)
+			commands_processed = append(commands_processed, s)
+		}
+	}
+
+	sort.Strings(commands_processed)
+
+	for _, value := range commands_processed {
+		fmt.Fprint(Stdout, value)
+	}
+
+	return nil
+}
+
 func (x *cmdAdviseSnap) Execute(args []string) error {
 	if len(args) > 0 {
 		return ErrExtraArgs
 	}
 
+	if x.DumpDb {
```

Comment:
We should add a test too, a spread test is fine I think. Something simple like:
```
--- a/tests/main/snap-advise-command/task.yaml
+++ b/tests/main/snap-advise-command/task.yaml
@@ -48,3 +48,6 @@ execute: |
 
     echo "Ensure advise on snap pkg name advise also works"
     snap advise-snap test-snapd-tools | MATCH "Tools for testing the snapd application"
+
+    echo "Ensure advice can dump the database"
+    snap advice-snap --dump-db | tr '\377' ':' | MATCH '^test-snapd-tools.echo:test-snapd-tools$'
 ```
But don't worry about it much, we (snapd team) can help with that :)